python/whylogs/api/store/query.py

```diff
@@ -1,36 +1,26 @@
-from abc import ABC
+from dataclasses import dataclass, field
 from datetime import datetime
-from typing import Optional
+from typing import Optional, Union
 
 
-class BaseQuery(ABC):
-    def __init__(self):
-        self.dataset_id = None
-
-
-class DateQuery(BaseQuery):
-    def __init__(
-        self,
-        dataset_id: str,
-        start_date: datetime,
-        end_date: Optional[datetime] = None,
-        segment_tag: Optional[str] = None,
-    ):
-
-        super().__init__()
-        self.dataset_id = dataset_id
-        self.start_date = start_date
-        self.end_date = end_date
-        self.segment_tag = segment_tag
+@dataclass
+class DateQuery:
+    dataset_id: str
+    start_date: datetime
+    end_date: Optional[datetime] = field(default=None)
+    segment_tag: Optional[str] = field(default=None)
```

Comment:
I think segment_tag falls into the same bucket. Looks like we can either make a dedicated query type for it or forever copy/paste it to every query type we make.

Reply:
removing `segment_tag` for now as we will need to flesh this out a little more on the following PRs